exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/UnitTestCoverageCollector.java

```diff
@@ -0,0 +1,113 @@
+package com.capitalone.dashboard.exec.collector;
+
+import com.capitalone.dashboard.exec.model.HygieiaSparkQuery;
+import com.capitalone.dashboard.exec.model.MetricType;
+import com.capitalone.dashboard.exec.model.MetricCollectionStrategy;
+import com.capitalone.dashboard.exec.model.CollectorItemMetricDetail;
+import com.capitalone.dashboard.exec.model.CollectorType;
+import com.capitalone.dashboard.exec.model.MetricCount;
+import com.capitalone.dashboard.exec.repository.PortfolioMetricRepository;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import scala.collection.JavaConversions;
+import scala.collection.mutable.WrappedArray;
+
+import java.util.List;
+import java.util.Collection;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Date;
+import java.util.HashMap;
+
+@Component
+public class UnitTestCoverageCollector extends DefaultMetricCollector {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UnitTestCoverageCollector.class);
+
+    @Autowired
+    public UnitTestCoverageCollector(PortfolioMetricRepository portfolioMetricRepository) {
+        super(portfolioMetricRepository);
+    }
+
+    @Override
+    protected MetricType getMetricType() {
+        return MetricType.UNIT_TEST_COVERAGE;
+    }
+
+    @Override
+    protected String getQuery() {
+        return HygieiaSparkQuery.STATIC_CODE_ANALYSIS_QUERY_ALL_COLLECTOR_ITEMS;
+    }
+
+    @Override
+    protected String getCollection() {
+        return "code_quality";
```

Comment:
There appears to be an inconsistency in the use of constants
That said the StaticCodeAnalysisCollector also doesnt use constants for MongoDB Collection names.
Perhaps an opportunity to fix the code base?